v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java

```diff
@@ -0,0 +1,494 @@
+    @Description(
+        "The Cloud Pub/Sub subscription to consume from."
+            + "The name should be in the format of "
+            + "projects/<project-id>/subscriptions/<subscriptions-name>")
+    @Validation.Required
+    String getInputSubscription();
+
+    void setInputSubscription(String inputSubscription);
+
+    @Description("The MongoDB database to push the Documents to.")
+    @Validation.Required
+    String getDatabase();
+
+    void setDatabase(String database);
+
+    @Description(
+        "The host addresses of the MongoDB"
+            + "Multiple addresses to be specified with a comma separated value e.g."
+            + "host1:port,host2:port,host3:port")
+    @Validation.Required
+    String getMongoDBUri();
+
+    void setMongoDBUri(String mongoDBUri);
+
+    @Description("The Collection in mongoDB to put documents to.")
+    @Validation.Required
+    String getCollection();
+
+    void setCollection(String collection);
+
+    @Description(
+            "The dead-letter table to output to within BigQuery in <project-id>:<dataset>.<table> "
+                    + "format.")
+    @Validation.Required
+    String getDeadletterTable();
+
+    void setDeadletterTable(String deadletterTable);
+
+    @Description("Batch size in number of documents. Default: 1000")
+    @Default.Long(1024)
+    Long getBatchSize();
+
+    void setBatchSize(Long batchSize);
+
+    @Description("Batch size in number of bytes. Default: 5242880 (5mb)")
+    @Default.Long(5242880)
+    Long getBatchSizeBytes();
+
+    void setBatchSizeBytes(Long batchSizeBytes);
+
+    @Description("Maximum Connection idle time")
+    @Default.Integer(60000)
+    int getMaxConnectionIdleTime();
+
+    void setMaxConnectionIdleTime(int maxConnectionIdleTime);
+
+    @Description("Specify if SSL is enabled")
+    @Default.Boolean(false)
+    Boolean getSslEnabled();
+
+    void setSslEnabled(Boolean sslEnabled);
+
+    @Description("Specify whether to ignore SSL certificate")
+    @Default.Boolean(false)
+    Boolean getIgnoreSSLCertificate();
+
+    void setIgnoreSSLCertificate(Boolean ignoreSSLCertificate);
+
+    @Description("Enable ordered bulk insertions")
+    @Default.Boolean(true)
+    Boolean getWithOrdered();
+
+    void setWithOrdered(Boolean withOrdered);
+
+    @Description("Enable invalidHostNameAllowed for ssl connection")
+    @Default.Boolean(false)
+    Boolean getWithSSLInvalidHostNameAllowed();
+
+    void setWithSSLInvalidHostNameAllowed(Boolean withSSLInvalidHostNameAllowed);
+  }
+
+  /**
+   * DoFn that will parse the given string elements as Bson Documents.
+   */
+  public static class ParseAsDocumentsFn extends DoFn<String, Document> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ParseAsDocumentsFn.class);
+
+    @ProcessElement
+    public void processElement(ProcessContext context) {
+      context.output(Document.parse(context.element()));
```

Comment:
Will this parsing always be successful? If not, we should put this in a try-catch block and use separate tags to catch these parsing failures (with metric-counters).

Reply:
The function will always get a valid json. Valid json will always get parsed as Bson documents

Reply:
Yes in the context of this pipeline. However, what happens when I attempt to use the public static class and pass a String (which is not a valid json)?